tests/test_digital.py

```diff
@@ -322,3 +325,34 @@ def test_dm(n_qubits: Tensor, batch_size: Tensor) -> None:
     dm = density_mat(state_cat)
     assert dm.size() == torch.Size([2**n_qubits, 2**n_qubits, batch_size])
     assert torch.allclose(dm, dm_proj)
+
+
+@pytest.mark.parametrize("operator", GATESET)
+def test_promote(operator: Tensor) -> None:
```

Comment:
```suggestion
def test_promote(operator: list) -> None:
```

Reply:
Done.

Reply:
`@pytest.mark.parametrize("operator", GATESET)`does ensure that the input is an element of `GATESET`. So it be rather:
```python3
@pytest.mark.parametrize("operator", GATESET)
def test_operator_product(operator: Primitive) -> None: